lib/src/widgets/settings_item.dart

```diff
@@ -82,10 +83,12 @@ class SettingsItem extends StatelessWidget {
     @required this.onTap,
     this.showSwitch = false,
     this.onSwitchChanged,
+    this.textColor,
   }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
+    if (this.textColor == null) this.textColor = CustomTheme.of(context).onSurface;
```

Comment:
Just as a hint, `this` most of the time isn't required. Only if there is also a local variable named like the class field we prepend this.